plugin/src/main/java/com/denizenscript/denizen/tags/core/TextTagBase.java

```diff
@@ -180,41 +183,67 @@ public TextTagBase() {
+            ListTag with = null;
+            MapTag translateMap = param.asType(MapTag.class, CoreUtilities.noDebugContext);
+            if (translateMap != null) {
+                ElementTag translateElement = translateMap.getRequiredObjectAs("key", ElementTag.class, attribute);
+                if (translateElement == null) {
+                    return null;
+                }
+                translateText = translateElement.asString();
+                fallback = translateMap.getElement("fallback");
+                with = translateMap.getObjectAs("with", ListTag.class, attribute.context);
             }
-            String translateText = attribute.getParam();
-
-            // <--[tag]
-            // @attribute <&translate[<key>].with[<text>|...]>
-            // @returns ElementTag
-            // @description
-            // Returns a special chat code that displays an autotranslated message.
-            // Optionally, specify a list of escaped text values representing input data for the translatable message.
-            // Be aware that missing 'with' values will cause exceptions in your console.
-            // For example: - narrate "<&translate[commands.give.success.single].with[32|<&translate[item.minecraft.diamond_sword].escaped>|<player.name.escaped>]>"
-            // Be warned that language keys change between Minecraft versions.
-            // Note that this is a magic Denizen tool - refer to <@link language Denizen Text Formatting>.
-            // -->
-            StringBuilder with = new StringBuilder();
-            if (attribute.startsWith("with", 2)) {
-                ListTag withList = attribute.contextAsType(2, ListTag.class);
-                attribute.fulfill(1);
-                for (String str : withList) {
-                    with.append(";").append(FormattedTextHelper.escape(EscapeTagUtil.unEscape(str)));
+            else {
+                BukkitImplDeprecations.translateLegacySyntax.warn(attribute.context);
+                translateText = param.toString();
+
+                // <--[tag]
+                // @attribute <&translate[<key>].with[<text>|...]>
+                // @returns ElementTag
+                // @deprecated Use '<&translate[key=<key>;with=<text>|...]>'.
+                // @description
+                // Deprecated in favor of <@link tag &translate>.
+                // -->
+                if (attribute.startsWith("with", 2)) {
+                    with = attribute.contextAsType(2, ListTag.class);
+                    attribute.fulfill(1);
+                }
+            }
+            StringBuilder output = new StringBuilder().append(ChatColor.COLOR_CHAR).append("[translate=").append(FormattedTextHelper.escape(translateText));
+            if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20) && fallback != null) {
+                output.append(';').append(ChatColor.COLOR_CHAR).append("fallback=").append(FormattedTextHelper.escape(fallback.asString()));

```

Comment:
injecting color_char _inside of_ the block here is problematic

Reply:
Updated to be a `MapTag` as discussed on Discord